AK-8  implementing CRUD operations

adamski83 opened the pull request.

adamski83 committed:
refactored user register with instance of axios

adamski83 committed:
cleaning axios requests and reorganization structure of application

adamski83 committed:
Merge pull request #5 from adamski83/main

Merge pull request #4 from adamski83/AK-6

adamski83 committed:
extracting registerUserApi to separate file

adamski83 committed:
Merge branch 'AK-6' of https://github.com/adamski83/boxes_app_front into AK-6

adamski83 committed:
added autentification process and login

adamski83 committed:
refactor: Extract registerUserApi to separate file, managing authorization with localStorage

adamski83 committed:
reviewing codbase after PR

adamski83 committed:
Merge branch 'main' into AK-6

adamski83 committed:
AK-8  implementing CRUD operations

adamski83 committed:
AK-8  implementing CRUD operations

adamski83 committed:
cleaning a mess

fengerkamil commented on src/components/dashboard/Dashboard.tsx:
```diff
@@ -23,7 +23,7 @@ const Dashboard = () => {
   const { mutate: addNewBox } = useAddNewBox({
     onSuccess: (data) => {
-      queryClient.invalidateQueries("boxes");
+      queryClient.invalidateQueries({ queryKey: ["boxes"] });
```
Dlatego właśnie polecałem utrzymywanie kluczy obok plików mutacji (tam gdzie mamy axiosa + useQuery/useMutation) aby tutaj zaimportować stałą BOXES i nie wypisywać wartości z palca :P

fengerkamil commented on src/components/dashboard/Dashboard.tsx:
```diff
@@ -70,7 +69,7 @@ const Dashboard = () => {
     <Container>
       <SearchBar onSearch={handleSearch} />
-      <form onSubmit={handleSubmit(onSubmit)}>
+      <form onSubmit={handleSubmit(onSubmit)} style={{ margin: 20 }}>
```
Lepiej dodać komponent <Box /> z materiala i nim opakować form dodając mu styl marginesu, przynajmniej zostanie przeniesiony do stylów, a ten tutaj zostanie dodany jako inline HTML  - więc źle

fengerkamil commented on src/components/dashboard/Dashboard.tsx:
```diff
@@ -88,8 +87,14 @@ const Dashboard = () => {
           control={control}
-          defaultValue={[0, 0, 0]}
-          render={({ field }) => <TextField {...field} label="New Dimension" />}
+          defaultValue="0,0,0"
+          render={({ field }) => (
+            <TextField
+              {...field}
+              label="Dimensions (comma separated)"
+              placeholder="e.g. 10,20,30"
+            />
+          )}
```
Może fajnie byłoby tutaj dodać walidację regexem czy tekst składa się z trzech liczb przedzielonych dwoma przecinkami?

fengerkamil commented on src/noteCard/NoteCard.tsx:
```diff
@@ -1,36 +1,159 @@
+  const { control, handleSubmit } = useForm<MockDataItem>({
+    defaultValues: item,
+  });
+  const [isEditing, setIsEditing] = useState<boolean>(true);
```
useState zazwyczaj trzymamy najwyżej w komponencie + nie musimy ustalać boolean jako typu, skoro ustalamy z palca na start true, TS sie domyśli sam.

fengerkamil commented on src/noteCard/NoteCard.tsx:
```diff
@@ -1,36 +1,159 @@
-          </Typography>
-        </CardContent>
-      </Card>
+      {isEditing ? (
```
Zbyt duży jest ten if i jego dwie wersje. Rozdzieliłbym to na dwa returny w tym pliku LUB na dwa komponenty dziecka

fengerkamil commented on src/noteCard/NoteCard.tsx:
```diff
@@ -1,36 +1,159 @@
+                  <DeleteOutlineIcon />
+                </IconButton>
+                <IconButton
+                  onClick={() => setIsEditing((isEditing) => !isEditing)}
```
Nie tworzymy funkcji inline w jSX, to zła praktyka

fengerkamil commented on src/noteCard/NoteCard.tsx:
```diff
@@ -1,36 +1,159 @@
+          <CardHeader
+            action={
+              <>
+                <IconButton onClick={() => useDelete(item._id)}>
```
Tutaj też możęmy uniknac inline funkcji. Lepiej zrobić tak:
onClick={prepareUseDelete(item._id));

a gdzieś wyżej: 
```
const prepareUseDelete = (id: string) => () => {
costamCostam(id);
}
```

Chodzi o to, że wywołujemy funkcję od razu przekazujac ID i ona nam zwraca kolejną funkcje, która już to ID pamięta:)

fengerkamil commented on src/noteCard/NoteCard.tsx:
```diff
@@ -1,36 +1,159 @@
+        </Card>
+      ) : (
+        <>
+          <Card key={item._id}>
+            <CardHeader
+              action={
+                <>
+                  <IconButton onClick={() => useDelete(item._id)}>
+                    <DeleteOutlineIcon />
+                  </IconButton>
+                  <IconButton
+                    onClick={() => setIsEditing((isEditing) => !isEditing)}
+                  >
+                    <EditIcon />
+                  </IconButton>
+                </>
+              }
+              title={item.name}
+            />
+            <CardContent>
+              <Typography variant="body2" color="textSecondary">
+                {/* {modifedDimension} */}
+              </Typography>
+              <Typography variant="body2" color="textSecondary">
+                {item.usage}
+              </Typography>
+              <Typography variant="body2" color="textSecondary">
+                {item.amount}
+              </Typography>
+            </CardContent>
+          </Card>
```
To nie praktycznie wszystko to samo co wyżej? Dodatkowo, zdestrukturyzowałbym obiekt item wyżej zeby nie powtarzać tutaj wszedzie notacji item.prop item.name item.usage

adamski83 committed:
AK-8 ,applied changes suggested in code review, commit fa97b27

adamski83 committed:
AK-8 ,applied changes suggested in code review, commit fa97b27

adamski83 committed:
builded app

adamski83 committed:
builded app

fengerkamil commented on src/App.tsx:
```diff
@@ -2,35 +2,57 @@ import "./App.css";
 function App() {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
```
nazwy breakpointów czyli sm, md, lg, xlg itd mogły by być zmiennymi gdzieś globalnie zdefiniowanymi

fengerkamil commented on src/App.tsx:
```diff
@@ -2,35 +2,57 @@ import "./App.css";
 import { SidebarMenu } from "./components/sidebar/SidebarMenu";
 import { Route, Routes } from "react-router-dom";
 import Home from "./components/home/Home";
+import MenuIcon from "@mui/icons-material/Menu";
 import Dashboard from "./components/dashboard/Dashboard";
 import Invoices from "./components/invoices/Invoices";
 import { PATHS } from "./urls/urls";
 import { Register } from "./components/register/Register";
 import { Login } from "./components/login/Login";
 import { Toaster } from "react-hot-toast";
 import { assignTokenIntoAPI } from "./services/assignTokenIntoAPI";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
+import { useTheme } from "@mui/material/styles";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import { Drawer, IconButton } from "@mui/material";
```
Fajnym sposobem na takie "niezorganizowane" importy jest dodanie wtyczki do prettiera i eslinta która sortuje nam według kategorii importy, możesz przynajmniej się rozejrzeć w internecie jak coś takiego działa. Na komercyjnych projektach tak pomieszane importy nie przechodzą w code-review

fengerkamil commented on src/components/dashboard/Dashboard.tsx:
```diff
@@ -23,31 +35,18 @@ const Dashboard = () => {
   const { mutate: addNewBox } = useAddNewBox({
     onSuccess: (data) => {
-      queryClient.invalidateQueries({ queryKey: ["boxes"] });
+      queryClient.invalidateQueries({ queryKey: [GET_BOXES] });
```
super

adamski83 committed:
AK-8  applayed CR sugestions:
extern file with breakpoinst,
avoid inline function declerating in JSX,
Strudcured import tree,

adamski83 committed:
AK-8  applayed CR sugestions:
extern file with breakpoinst,
avoid inline function declerating in JSX,
Strudcured import tree,

fengerkamil commented on src/components/sidebar/SidebarMenu.tsx:
```diff
@@ -1,42 +1,43 @@
     {
+      to: "",
       className: "menu1",
       icon: <MenuRoundedIcon onClick={onClose} />,
```
Powinnismy uzyc tego onClicka w miejscu gdzie iterujemy po tej tablicy na elemencie np MenuItem

fengerkamil commented on src/components/sidebar/SidebarMenu.tsx:
```diff
@@ -1,42 +1,43 @@
     {
       to: "/login",
-      icon: <LogoutRoundedIcon />,
+      icon: <LogoutRoundedIcon onClick={onClose} />,
```
Poniżej masz menuItems map i przypisujesz bo SubMenu index (nie moge dodac komentarza w niezmienionym kodzie chyba)

Tutaj artykuł dlaczego index jako key jest złym wyborem:
https://medium.com/react-in-the-real-world/why-you-should-never-ever-use-indices-as-react-keys-8329472170be

adamski83 committed:
Merge branch 'main' of https://github.com/adamski83/boxes_app_front

adamski83 committed:
Merge branch 'main' into AK-8

adamski83 committed:
Merge branch 'AK-8' of https://github.com/adamski83/boxes_app_front into AK-8
